zerver/lib/rate_limiter.py

```diff
@@ -110,8 +111,9 @@ def rules(self) -> List[Tuple[int, int]]:
 
 
 class RateLimitedUser(RateLimitedObject):
-    def __init__(self, user: UserProfile, domain: str = "api_by_user") -> None:
-        self.user = user
+    def __init__(self, user_id: int, rate_limits: str, domain: str = "api_by_user") -> None:
+        self.user_id = user_id
+        self.rate_limits = rate_limits
```

Comment:
The constructor’s API doesn’t need to change.

```suggestion
     def __init__(self, user: UserProfile, domain: str = "api_by_user") -> None:
        self.user_id = user.id
        self.rate_limits = user.rate_limits
```

Reply:
OK